libdrgn/program.c

```diff
@@ -995,3 +1001,64 @@ drgn_program_member_info(struct drgn_program *prog, struct drgn_type *type,
 	ret->bit_field_size = member->bit_field_size;
 	return NULL;
 }
+
+struct drgn_string_symbol {
+	const char *name;
+	struct drgn_symbol *symbol;
+};
+
+static int drgn_dwarf_module_iterate_sym_table(
+	Dwfl_Module *dwfl_module,
+	void **userdatap,
+	const char *module_name,
+	Dwarf_Addr base, void *arg)
+{
+	GElf_Sym elf_sym;
+	GElf_Addr elf_addr;
+	struct drgn_string_symbol *sym;
+	int symtab_offset, symtab_len;
+
+	sym = arg;
+	symtab_len = dwfl_module_getsymtab(dwfl_module);
+	symtab_offset = dwfl_module_getsymtab_first_global(dwfl_module);
+
+	for (int i = symtab_offset; i < symtab_len; i++) {
+		const char *name = dwfl_module_getsym_info(
+			dwfl_module, i, &elf_sym, &elf_addr, NULL, NULL, NULL);
+
+		if (strcmp(sym->name, name) == 0) {
+			sym->symbol->name = name;
+			sym->symbol->address = elf_addr;
+			sym->symbol->size = elf_sym.st_size;
+			return DWARF_CB_ABORT;
```

Comment:
I don't like this method of signalling success on ABORT, but I'm not sure if there's another way to cleanly get the exit status from the caller. Would you prefer setting a field in `sym` to determine success instead?